### ISSUE TITLE
Conceal level for Obsidian links in front matter

### ISSUE THREAD

**Issue by jaymorelli96** [epwalsh/obsidian.nvim#282]
### 🐛 Describe the bug

Not sure if this is a problem with this plugin, my treesitter or my configuration.

For some reason the link in the front matter doesn't conceal the last ]
![image](https://github.com/epwalsh/obsidian.nvim/assets/61237619/5d462a72-48d9-4559-96c5-7ea669f82815)

![image](https://github.com/epwalsh/obsidian.nvim/assets/61237619/b07633f7-bd11-4dcd-8733-7f0f6831c416)

### Versions

```nvim --version```
```bash
NVIM v0.9.1
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc  -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wvla -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -fno-common -Wno-unused-result -Wimplicit-fallthrough -fdiagnostics-color=auto -fstack-protector-strong -DUNIT_TESTING -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/usr/include/luajit-5_1-2.1 -I/usr/include -I/usr/include/lua5.1 -I/home/abuild/rpmbuild/BUILD/neovim-0.9.1/build/src/nvim/auto -I/home/abuild/rpmbuild/BUILD/neovim-0.9.1/build/include -I/home/abuild/rpmbuild/BUILD/neovim-0.9.1/build/cmake.config -I/home/abuild/rpmbuild/BUILD/neovim-0.9.1/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

```nvim --headless -c 'lua require("obsidian").info()' -c q```
```bash
[obsidian.nvim (v2.4.0)] Commit SHA: 319f06024cdde97e0c2c3a836dff49b9f0d5f3bd
[plenary.nvim] Commit SHA: 55d9fe89e33efd26f532ef20223e5f9430c8b0c0
[nvim-cmp] Commit SHA: 538e37ba87284942c1d76ed38dd497e54e65b891
[telescope.nvim] Commit SHA: 776b509f80dd49d8205b9b0d94485568236d1192
ripgrep 13.0.0
```

**Comment by epwalsh** [comment]
Hey @jaymorelli96, fixed in #284 :)

**Comment by jaymorelli96** [comment]
Perfect, thanks!
